lib/puppet/http/resolver/srv.rb

```diff
@@ -5,7 +5,7 @@ def initialize(domain: srv_domain, dns: Resolv::DNS.new)
   end
 
   def resolve(session, name, &block)
-    # This assumes the route name is the same as the DNS SRV name
+    # This assumes the service name is the same as the DNS SRV name
     @delegate.each_srv_record(@srv_domain, name) do |server, port|
       yield session.create_service(name, server, port)
```

Comment:
It is a little strange to me how we have to pass this through so many levels of the code. It goes from here to session.rb to service.rb to ca.rb. I don't think there's anything to be done, but it's just something I was thinking about.

Reply:
That's fair. It's here because the session has a reference to the client, and we need to pass that to the service: https://github.com/puppetlabs/puppet/blob/b19017727e9f3ebc84fd276c7ba002e094714f21/lib/puppet/http/session.rb#L37

The service's `get_certificate` method can then `@client.get` to make the actual request.

Alternatively, the client could be passed to each resolver when the latter is created, eg. `SettingsResolver.new(@client)`, but then it creates a circular reference, as the client creates/owns all of the resolvers, and each of the resolvers knows about the client. As a result, we can't create the resolvers as a stand alone object and inject that dependency into the client. We're not doing that now, but I want to in the near future.